src/bin/hive_chain.rs

```diff
@@ -67,10 +67,15 @@ async fn main() -> eyre::Result<()> {
             let chain_id = transaction.chain_id().ok_or(eyre!("Failed to recover chain id"))?;
             let starknet_tx = to_starknet_transaction(&transaction, chain_id, signer, u64::MAX)?;
 
+            let nonce = match &starknet_tx {
```

Comment:
I think starknet tx has a getter for nonce

Reply:
Unfortunatly not :/